frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/FetchBackgroundable.java

```diff
@@ -28,15 +26,14 @@ public class FetchBackgroundable extends Task.Backgroundable {
   /** Use as {@code remoteName} when referring to the local repository. */
   public static final String LOCAL_REPOSITORY_NAME = ".";
 
-  public FetchBackgroundable(Project project,
+  public FetchBackgroundable(
       GitRepository gitRepository,
       String remoteName,
       String refspec,
       String taskTitle,
       @Untainted String failureNotificationText,
       String successNotificationText) {
-    super(project, taskTitle, /* canBeCancelled */ true);
-    this.project = project;
+    super(gitRepository.getProject(), taskTitle, /* canBeCancelled */ true);
```

Comment:
A high-level sanity check... obviously the result of `getProject()` is fixed/immutable for the given instance of `GitRepository` right? 😅

Reply:
yup,
the method `getProject` is `@NotNull` in the `Repository` interface, in the implementation `RepositoryImpl` and as a field:
<img width="525" alt="image" src="https://user-images.githubusercontent.com/19799111/203785517-98a5fd6c-b2e3-4c54-b015-b8e7b08ae89e.png">